src/front/domain_socket.rs

```diff
@@ -105,7 +106,12 @@ impl Listen for DomainSocketListener {
                     format_error!("Failed to set stream as blocking", err);
                     None
                 } else {
-                    Some(Box::from(stream))
+                    Some(Connection {
+                        stream: Box::new(stream),
+                        // TODO: when possible, we want to replace this with the (uid, gid, pid)
+                        // triple for peer credentials.
+                        metadata: ConnectionMetadata::NoMetadata,
```

Comment:
It's not a conversation for here, just a random note, but I'd say that any IPC can/should have associated metadata - e.g. if we use TCP sockets, we'll have the peer IP address and port which could be used in the authentication.

Reply:
ACK 🙂